src/lib/snark_pool/snark_pool.ml

```diff
@@ -59,24 +81,72 @@ end) :
     let fee (t : t) = t.fee
   end
 
-  type t = Priced_proof.t Work.Table.t [@@deriving sexp, bin_io]
+  type t =
+    { snark_table: Priced_proof.t Work.Table.t
+    ; mutable ref_table: int Work.Table.t option }
```

Comment:
We shouldn't serialize the ref_table if possible. Ideally it could just be None when we serialize or something.
If not, we should at least set it to None at the very beginning of listen_to_frontier_broadcast_pipe

Reply:
I'm doing both. :-)